library/ssl_cli.c

```diff
@@ -307,29 +307,35 @@ static int ssl_write_supported_elliptic_curves_ext( mbedtls_ssl_context *ssl,
 {
     unsigned char *p = buf;
     unsigned char *elliptic_curve_list = p + 6;
+    const size_t curve_list_size = (size_t) ( end - p - 6 );
     size_t elliptic_curve_len = 0;
```

Comment:
Pre-existing defect worth fixing along the way: We don't check whether `end - p >= 6`.